src/Apps/Artwork/Components/ArtworkImageBrowser/__tests__/ArtworkActionsSaveButtonV2.jest.tsx

```diff
@@ -0,0 +1,272 @@
+import { fireEvent, screen } from "@testing-library/react"
+import { ArtworkActionsSaveButtonV2FragmentContainer } from "Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButtonV2"
+import { AppToasts } from "Apps/Components/AppToasts"
+import { ManageArtworkForSavesProvider } from "Components/Artwork/ManageArtworkForSaves"
+import { SaveArtwork } from "Components/Artwork/SaveButton/SaveArtworkMutation"
+import { MockBoot } from "DevTools/MockBoot"
+import { setupTestWrapperTL } from "DevTools/setupTestWrapper"
+import { graphql } from "react-relay"
+import { ArtworkActionsSaveButtonV2_Test_Query } from "__generated__/ArtworkActionsSaveButtonV2_Test_Query.graphql"
+
+jest.unmock("react-relay")
+jest.mock("Components/Artwork/SaveButton/SaveArtworkMutation")
+
+describe("ArtworkActionsSaveButtonV2", () => {
+  const mockSaveArtwork = SaveArtwork as jest.Mock
+
+  beforeEach(() => {
+    mockSaveArtwork.mockImplementation(() => ({}))
+  })
+
+  const { renderWithRelay } = setupTestWrapperTL<
+    ArtworkActionsSaveButtonV2_Test_Query
+  >({
+    Component: props => {
+      if (!props.artwork) {
+        return null
+      }
+
+      return (
+        <MockBoot context={{ user: { id: "percy-z" } }}>
+          <AppToasts accomodateNav={false} />
+
+          <ManageArtworkForSavesProvider>
+            <ArtworkActionsSaveButtonV2FragmentContainer
+              artwork={props.artwork}
+            />
+          </ManageArtworkForSavesProvider>
+        </MockBoot>
+      )
+    },
+    query: graphql`
+      query ArtworkActionsSaveButtonV2_Test_Query @relay_test_operation {
+        artwork(id: "artworkID") {
+          ...ArtworkActionsSaveButtonV2_artwork
+        }
+      }
+    `,
+  })
+
+  it("should display `Save` label", () => {
+    renderWithRelay({
+      Artwork: () => unsavedArtwork,
+    })
+
+    expect(screen.getByText("Save")).toBeInTheDocument()
+  })
+
+  it("should display `Watch lot` label", () => {
+    renderWithRelay({
+      Artwork: () => unsavedAuctionArtwork,
+    })
+
+    expect(screen.getByText("Watch lot")).toBeInTheDocument()
+  })
+
+  describe("Save flow", () => {
+    beforeEach(() => {
+      mockSaveArtwork.mockImplementation(args => ({
+        saveArtwork: {
+          artwork: {
+            is_saved: true,
```

Comment:
Should we also change the `is_saved` in the `SaveArtworkMutation` to match the [Artwork](https://github.com/artsy/force/pull/12132/files#diff-98dd7020596cdf678daa470d9dbfa754306be2db50b9f460c20e7b8064c5b179R63) query property (asking mostly to make sure that when we update via the mutation the store gets updated accordingly )

Reply:
in this case, we need to return `is_saved` because such a field is returned by the `SaveArtwork` mutation [here](https://github.com/artsy/force/blob/606283dd2ba0dd3a68d7878a78c7419977e017f6/src/Components/Artwork/SaveButton/SaveArtworkMutation.ts#L22).

I understand that there is too much confusion due to the fact that somewhere `isSaved` is returned, somewhere `is_saved'. I plan to do this next week. sorry for the confusion 😅

Reply:
Renamed here: https://github.com/artsy/force/pull/12161